mlos_core/mlos_core/tests/optimizers/optimizer_test.py

```diff
@@ -197,25 +203,54 @@ def objective(point: pd.DataFrame) -> pd.Series:
         "special_param_values": None,
         "max_unique_values_per_param": None,
     }
+
+    # Make some adjustments to the kwargs for the optimizer and LlamaTuned
+    # optimizer for debug/testing.
+
+    # if optimizer_type == OptimizerType.SMAC:
+    #    # Allow us to override the number of random init samples.
+    #    kwargs['max_ratio'] = 1.0
+    optimizer_kwargs = deepcopy(kwargs)
+    llamatune_optimizer_kwargs = deepcopy(kwargs)
+    # if optimizer_type == OptimizerType.SMAC:
+    #    optimizer_kwargs['n_random_init'] = 20
+    #    llamatune_optimizer_kwargs['n_random_init'] = 10
+
     llamatune_optimizer: BaseOptimizer = OptimizerFactory.create(
         parameter_space=input_space,
         optimizer_type=optimizer_type,
-        optimizer_kwargs=kwargs,
+        optimizer_kwargs=llamatune_optimizer_kwargs,
         space_adapter_type=SpaceAdapterType.LLAMATUNE,
         space_adapter_kwargs=space_adapter_kwargs,
     )
     # Initialize an optimizer that uses the original space
     optimizer: BaseOptimizer = OptimizerFactory.create(
         parameter_space=input_space,
         optimizer_type=optimizer_type,
-        optimizer_kwargs=kwargs,
+        optimizer_kwargs=optimizer_kwargs,
     )
     assert optimizer is not None
-
     assert llamatune_optimizer is not None
     assert optimizer.optimizer_parameter_space != llamatune_optimizer.optimizer_parameter_space
 
-    for _ in range(num_iters):
+    opt_n_random_init = 0
+    llamatune_n_random_init = 0
+    if 'n_random_init' in kwargs:
+        opt_n_random_init = int(kwargs['n_random_init'])
+    if optimizer_type == OptimizerType.SMAC:
+        assert isinstance(optimizer, SmacOptimizer)
+        assert isinstance(llamatune_optimizer, SmacOptimizer)
+        # pylint: disable=protected-access
+        opt_n_random_init = optimizer.base_optimizer._initial_design._n_configs
+        llamatune_n_random_init = llamatune_optimizer.base_optimizer._initial_design._n_configs
```

Comment:
maybe just implement a property `SmacOptimizer.n_configs`? mypy should already know that `optimizer` is of the right type, so we can write `opt_n_random_init = optimizer.n_configs`